commons/src/main/java/org/sonarsource/analyzer/commons/ProgressReport.java

```diff
@@ -92,14 +103,20 @@ public synchronized void nextFile() {
   public synchronized void stop() {
     success = true;
     thread.interrupt();
+    join();
   }
 
   public synchronized void cancel() {
     thread.interrupt();
+    join();
   }
 
-  public void join() throws InterruptedException {
-    thread.join();
+  private void join() {
+    try {
+      thread.join();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
```

Comment:
These two lines are apparently not covered. Would it be possible to try to check them?
I feel it's important as it propagates the interruption.

Reply:
Not very simple to test, but here it is.